docs: Improve clarity of next('route') usage in Route Handlers section

yelishgiri opened the pull request:
<!--
Thank you for your pull request. Please provide a description and 
note the Certificate of Origin below. 

-->

Description:
This PR adds a concrete example demonstrating the use of next('route') in the “Route handlers” section of the Express routing documentation.

Currently, the docs mention next('route') but don’t show a simple, self-contained example, which could make it unclear for new users. This change:

Adds a GET /user/:id example that uses next('route') to skip to the next matching route.
Includes a short bullet-point explanation of the behavior for two different request paths (/user/5 and /user/0).

Before:
The section explains next('route') but doesn’t show a minimal code sample demonstrating it in action.

After:
The section now contains:
```
app.get('/user/:id', (req, res, next) => {
  if (req.params.id === '0') {
    return next('route')
  }
  res.send(`User ${req.params.id}`)
})

app.get('/user/:id', (req, res) => {
  res.send('Special handler for user ID 0')
})

```

In this example:
GET /user/5 → handled by first route → sends "User 5"
GET /user/0 → first route calls next('route'), skipping to the next matching /user/:id route

Why this change improves the docs:

Makes the next('route') behavior obvious without requiring guesswork.
Matches the style and tone of existing Express documentation.
Adds a practical example to complement the textual explanation.


<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

yelishgiri committed:
docs(route-handlers): add example for next('route') usage

bjohansebas commented on en/guide/routing.md:
```diff
@@ -245,6 +245,26 @@ In Express 4.x, <a href="https://github.com/expressjs/express/issues/2495">the `
 
 You can provide multiple callback functions that behave like [middleware](/{{ page.lang }}/guide/using-middleware.html) to handle a request. The only exception is that these callbacks might invoke `next('route')` to bypass the remaining route callbacks. You can use this mechanism to impose pre-conditions on a route, then pass control to subsequent routes if there's no reason to proceed with the c ...
 
+You can use `next('route')` to skip the rest of a route's callbacks and pass control to the next route. For example:
```
These two sentences repeat content, whereas if the message can be improved and shown through the example, we shouldn’t repeat the content that was already mentioned earlier.

yelishgiri replied:
Thanks for pointing that out. I’ve removed the second sentence so the explanation is given only once, with the example showing the behavior directly. This should make the section more concise and avoid redundancy.

yelishgiri committed:
docs(route-handlers): remove redundant sentence in next('route') section

Removed a duplicate explanation of next('route') to avoid redundancy.  
The section now introduces the concept once and flows directly into the example,  
making the docs more concise and easier to read.

yelishgiri committed:
Merge branch 'gh-pages' into docs-add-next-route-example

bjohansebas reviewed (approved):
LGTM, thanks
